core/embed/rust/src/micropython/list.rs

```diff
@@ -92,6 +100,22 @@ impl List {
             core::slice::from_raw_parts_mut(items_ptr, len)
         }
     }
+
+    pub fn get(&self, index: usize) -> Result<Obj, Error> {
```

Comment:
the unsafe ones are unsafe :) `as_slice` is unsafe because the slice does not necessarily uphold the invariants that Rust promises during its whole lifetime.

`get` is safe because it is using `as_slice` exactly as safely as can be done, and the result _does_ uphold the relevant invariants.

Reply:
N.b.: marking function `unsafe` is completely independent of whether it contains any `unsafe` blocks inside, and in some weird cases a function can be `unsafe` despite having no `unsafe` blocks.

Marking a function `unsafe` is information for the caller: be careful and explicit (i.e., use an `unsafe` block) when calling this, because you _could_ be doing it wrong. Safe functions are, or should be, safe to call without thinking too much about it.